src/storage/errors.rs

```diff
@@ -16,6 +17,8 @@ use txn_types::{KvPair, TimeStamp};
 
 quick_error! {
     #[derive(Debug)]
+    /// To help simplify pattern matching of storage related errors, and to have
```

Comment:
I think it's better the state "what is it" instead of "why is it". Like "Detailed errors for storage operations."

Reply:
Added the What is It part at the start and edit the rest of the docs a bit.